### PR TITLE
♻️ Prometheus Monitoring Stack Refactoring 

### PR DESCRIPTION
**Pull request by mrnicegyu11**
*This is 2/3 from split huge PR https://github.com/ITISFoundation/osparc-ops-environments/pull/117*

This PR brings:
- ✨ Adds j2 templating for postgres- and redis-prometheus-exporter
- ✨ Adds postgres-query-exporter `adonato/query-exporter:2.8.3`
- ✨ Adds prometheus metrics for number of s4l-lite users and total number of users
- ♻️ Harmonize prometheus configuration: Always scrape potentially existing master, staging and production stacks
- ♻️ Harmonizes monitoring stack: Remove references to dalco, minio
- 🔨 Removes minio-scrapers in prometheus configuration

👷👷👷 `osparc-ops-deployment-configuration` changes 👷👷👷:
- 🔨 REMOVE - From now on unused: `POSTGRES_EXPORTER_DATA_SOURCE_NAME`
- 🔨 REMOVE - From now on unused: `REDIS_EXPORTER_DATA_SOURCE_NAME`
- 🔨 REMOVE - From now on unused: `PRODUCTION_POSTGRES_EXPORTER_DATA_SOURCE_NAME`
- 🔨 REMOVE - From now on unused: `STAGING_POSTGRES_EXPORTER_DATA_SOURCE_NAME`
- 🔨 REMOVE - From now on unused: `PRODUCTION_REDIS_EXPORTER_DATA_SOURCE_NAME`
- 🔨 REMOVE - From now on unused: `STAGING_REDIS_EXPORTER_DATA_SOURCE_NAME`
- ♻️ RENAME: `PROMETHEUS_RETENTION` --> `MONITORING_PROMETHEUS_RETENTION`
- ♻️ RENAME: `PROMETHEUS_PORT` --> `MONITORING_PROMETHEUS_PORT`
- ♻️ RENAME: `PROMETHEUS_CEPH_TARGETS` --> `MONITORING_PROMETHEUS_CEPH_TARGETS`
- ✨ ADD: `MONITORING_PROMETHEUS_PGSQL_GID_MONITORED`: Comma-sperated list of ints. GIDs in the pgSQL database for which the number of associated users will be monitored and exported to prometheus. Example: `MONITORING_PROMETHEUS_PGSQL_GID_MONITORED="135745,111662,"`
- ✨ ADD: `MONITORED_STACK_NAMES`: Comma-seperated list of strings. Names of stacks for which a redis/pgSQL are assumed to be present and that will be monitored. Example: `MONITORED_STACK_NAMES="production,"`

🚧🚧🚧 Necessary when rolling out 🚧🚧🚧:
- [ ] Adjust env-vars

### CODE REVIEW

**Comment by mrnicegyu11** [comment]
FYI: This was successfully tested on aws-prod and on my local deployment

**Review by pcrespov — COMMENTED** [review]
:+1:

File: scripts/migration/db-restore.bash [hunk 1]
```diff
@@ -38,7 +38,5 @@ if [ "$yn" = "y" ]; then
     -v "$(pwd)":/var/pgdata \
     jbergknoff/postgresql-client postgresql://"${POSTGRES_USER}":"${POSTGRES_PASSWORD}"@"${POSTGRES_HOST}":"${POSTGRES_PORT}"/postgres \
     -c "DROP DATABASE ${POSTGRES_DB};" -c "CREATE DATABASE ${POSTGRES_DB};" \
```
Comment (sanderegg) [inline scripts/migration/db-restore.bash:40]: removing this will break the NIH metrics dashboard right?

File: services/monitoring/Makefile [hunk 2]
```diff
@@ -28,7 +28,7 @@ up: .init .env config.prometheus ${TEMP_COMPOSE} ## Deploys or updates current s
 	@docker stack deploy --with-registry-auth --prune --compose-file ${TEMP_COMPOSE} $(STACK_NAME)
 
 .PHONY: up-local
-up-local: .init .env config.prometheus ${TEMP_COMPOSE}-local ## Deploys or updates current stack "$(STACK_NAME)". If MONITORED_NETWORK is not specified, it will create an attachable network
+up-local: .init .env config.prometheus.ceph.simcore ${TEMP_COMPOSE}-local ## Deploys or updates current stack "$(STACK_NAME)". If MONITORED_NETWORK is not specified, it will create an attachable network
```
Comment (sanderegg) [inline services/monitoring/Makefile:31]: why ceph for local?

**Review by sanderegg — APPROVED** [review]
very nice. maybe just check the my comments. thanks!

File: scripts/migration/db-restore.bash (hunk 1 above)
Comment (mrnicegyu11) [reply scripts/migration/db-restore.bash:40]: yes, but only until PR 3/3 fixes them by using different metrics ;) Sorry about this, it goes together with changes in the NIH metrics dashboard yaml file in the other PR

File: services/monitoring/Makefile (hunk 2 above)
Comment (mrnicegyu11) [reply services/monitoring/Makefile:31]: good point this can leave away ceph, I will adress this

**Pull request merged**